src/main/java/com/faforever/client/filter/ChatUserFilterController.java

```diff
@@ -0,0 +1,115 @@
+package com.faforever.client.filter;
+
+import com.faforever.client.chat.ChatChannelUser;
+import com.faforever.client.chat.ListItem;
+import com.faforever.client.domain.LeaderboardBean;
+import com.faforever.client.domain.PlayerBean;
+import com.faforever.client.game.PlayerStatus;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.leaderboard.LeaderboardService;
+import com.faforever.client.player.Country;
+import com.faforever.client.player.CountryFlagService;
+import com.faforever.client.theme.UiService;
+import com.faforever.client.util.RatingUtil;
+import javafx.util.StringConverter;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class ChatUserFilterController extends AbstractFilterController<ListItem> {
+
+  private final CountryFlagService countryFlagService;
+  private final LeaderboardService leaderboardService;
+
+  private final static int MIN_RATING = -1000;
+  private final static int MAX_RATING = 4000;
+
+  private final StringConverter<PlayerStatus> playerStatusConverter = new StringConverter<>() {
+    @Override
+    public String toString(PlayerStatus object) {
+      return i18n.get(object.getI18nKey());
+    }
+
+    @Override
+    public PlayerStatus fromString(String string) {
+      throw new UnsupportedOperationException("Not supported");
+    }
+  };
+
+  private final StringConverter<Country> countryConverter = new StringConverter<>() {
+    @Override
+    public String toString(Country object) {
+      return String.format("%s [%s]", object.displayName(), object.code());
+    }
+
+    @Override
+    public Country fromString(String string) {
+      throw new UnsupportedOperationException("Not supported");
+    }
+  };
+
+  private final StringConverter<LeaderboardBean> leaderboardConverter = new StringConverter<>() {
+    @Override
+    public String toString(LeaderboardBean object) {
+      String rating = i18n.getOrDefault(object.getTechnicalName(), object.getNameKey());
+      return i18n.get("leaderboard.rating", rating);
+    }
+
+    @Override
+    public LeaderboardBean fromString(String string) {
+      throw new UnsupportedOperationException("Not supported");
+    }
+  };
```

Comment:
Just a comment but I feel like I should really just write a functioninterface that is a one way string converter for how much we use this pattern

Reply:
They are needed for `view.setConverter(StringConverter)` method. The functional interface is not suitable.